src/Project/Job.ts

```diff
@@ -16,15 +16,19 @@
 
 import {ToolArgs} from './ToolArgs';
 
+export type JobCallback = {
```

Comment:
When I wrorte this code, I didn't consider JobCallback's usage enough. In the future, this type could be different from this.

Reply:
And extracting it as this type looks very good.